cpp/include/cudf/stream_compaction.hpp

```diff
@@ -208,9 +208,11 @@ std::unique_ptr<table> apply_boolean_mask(
  * @brief Choices for drop_duplicates API for retainment of duplicate rows
  */
 enum class duplicate_keep_option {
-  KEEP_FIRST = 0,  ///< Keeps first duplicate row and unique rows
-  KEEP_LAST,       ///< Keeps last  duplicate row and unique rows
-  KEEP_NONE        ///< Keeps only unique rows are kept
+  KEEP_FIRST = 0,  ///< Keeps first duplicate element and unique elements
+  KEEP_LAST,       ///< Keeps last duplicate element and unique elements
+  KEEP_ANY_ONE,  ///< Keeps one duplicate element at an undefined position and unique elements (this
+                 ///< option may not be supported in certain operations)
```

Comment:
I don't understand why this option is necessary. `KEEP_LAST/KEEP_FIRST` are valid results for `KEEP_ANY`, so why expose another option.

Reply:
Also, if it is indeed necessary, it should just be called `KEEP_ANY`.

Reply:
Also, the existing `drop_duplicates` API should be updated to support this option by just rerouting to the `KEEP_FIRST` or `KEEP_LAST` code path (which brings me to my original question as to why we'd want a `KEEP_ANY` option in the first place).

Reply:
Curently `sort_lists` internally calls both `cub` or `segmented_sort_order` for sorting the list entries, and there is no guarantee that the result will be of stable sort (maybe currently stable, but nothing is guaranteed).

Thus, in order to enforce `KEEP_LAST/KEEP_FIRST`, I need to add extra data structures to maintain entry order after sorted. If we want just `KEEP_ANY_ONE`, such extra data is not needed---just do (unstable/default) sorting and unique copy.

Reply:
Hah, I also realize that `drop_duplicates` calls unstable sort too. Thus, it may produce wrong result if specifying `KEEP_LAST/KEEP_FIRST` (didn't verify this).